test/axon_test.exs

```diff
@@ -477,6 +477,20 @@ defmodule AxonTest do
     end
   end
 
+  @global_pooling_layers [:global_average_pool, :global_max_pool, :global_lp_pool]
+
+  describe "global pooling" do
+    test "works with options" do
+      for pool <- @global_pooling_layers do
+        assert %Axon{name: "pool", output_shape: {nil, 1}} =
+                 apply(Axon, pool, [
+                   Axon.input({nil, 1, 28, 28}),
+                   [keep_axes: false, name: "pool"]
+                 ])
+      end
+    end
+  end
```

Comment:
This doesn't test it actually works, right? It may be worth changing those tests later on to execute, even if we don't assert on the result, but to guarantee Nx+EXLA compiles it. :)

Reply:
Correct, I have decided that to handle this here and with other layers I am going to introduce a `compiler_test.exs` file. So `axon_test` will assert the model is being constructed and inspected correctly and compiler test will assert on results